From 39ea3f4a82be9d745fd516d1cff001a3e5dbf7cc Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 9 Oct 2018 17:26:43 +0800
Subject: [PATCH 4806/5242] MLK-19819-2 Revert "MLK-19318 drm/imx: lcdif:
 avoid log print when no device attached"

commit  fb0fb068138babe19bc34bf4ca526fb9a484e67e from
https://source.codeaurora.org/external/imx/linux-imx.git

This reverts commit 05d335b07bab856b34317acbe3e3b9d9f58946ce, since the
commit e07309cfdc567623a3f0cde6b79b972910248152 (MLK-19819-1 drm/imx:
lcdif: bypass atomic check when CRTC is disabled) can cover the function
that commit 05d335b07bab856b34317acbe3e3b9d9f58946ce can provide.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/lcdif/lcdif-crtc.c |    7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
index 7d83a04..216baf1 100644
--- a/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
+++ b/drivers/gpu/drm/imx/lcdif/lcdif-crtc.c
@@ -100,13 +100,6 @@ static int lcdif_crtc_atomic_check(struct drm_crtc *crtc,
 	if (!state->enable)
 		return 0;
 
-	/* return directly when no devices attached
-	 * to LCDIF to avoid below error messsage to
-	 * make noises in this case.
-	 */
-	if (!imx_crtc_state->bus_format)
-		return -EINVAL;
-
 	/* check the requested bus format can be
 	 * supported by LCDIF CTRC or not
 	 */
-- 
1.7.9.5

